File: java/Raindrops.java
class RaindropConverter {

    String convert(int number) {
       StringBuilder str = new StringBuilder("");
       if (number % 3 == 0) str.append("Pling");
       if (number % 5 == 0) str.append("Plang");
       if (number % 7 == 0) str.append("Plong");
       String result = str.toString();       
       return (result.equals("")) ?
         String.valueOf(number) :
         result; 
    }

}

File: c/raindrops/raindrops.c
#include "raindrops.h"
#include <string.h>
#include <stdio.h>

void convert(char* result, int drops) {
    if (!(drops % 3)) strcat(result, "Pling");
    if (!(drops % 5)) strcat(result, "Plang");
    if (!(drops % 7)) strcat(result, "Plong");  
    if (!strlen(result)) sprintf(result, "%d", drops);  
}

